ZEP000-wf-stream-processing.md

```diff
@@ -0,0 +1,1421 @@
+  events
+
+### Service Task
+
+Command to Process | Written Events and Follow-up Commands
+---|---
+ServiceTask:Activate_Element | ServiceTask:Element_Activating, [Variable:Created], ServiceTask:Element_Activated, Job:Created
+JobBatch:Activate | JobBatch:Activated 
+Job:Complete | Job:Completed, ServiceTask:Complete_Element
+ServiceTask:Complete_Element | ServiceTask:Element_Completing, [Variable:Created], [Variable:Updated], ServiceTask:Element_Completed, Sequence_Flow_Taken, EndEvent:Activate_Element,
+
+Notes:
+* if the service task has an input variable mapping then it produces one or more `Variable:Created` events for the new local variables
+* if the job is completed with variables then these variables will be stored in the state until they are used for the output mapping (i.e. temporal/transient variables). They must therefore also be written into the `Job:Completed` event.
+* if the service task has an output variable mapping, or the job is completed with variables, then it produces one or more `Variable:Created`/`Variable:Updated` events for the new and updated variables of the workflow instance or scope
+* the order of writing follow-up events matters. Both `ServiceTask:Element_Activated` and `Variable:Created` need to be replayed before `Job:Created`.
+
+On processing:
+* the `ServiceTask:Activate_Element` command transitions the service task
+  element to activated, may create variables for input mapping and creates a job
+* the `JobBatch:Activate` command activates a batch of jobs
+* the `Job:Complete` command completes a job
+* the `ServiceTask:Complete_Element` command transitions the service task
+  element to completed, may create and/or update variables for output mapping
+  and marks the sequence flow as taken
+
+On replay:
+* the `ServiceTask:Element_Activating` event transitions the service task element to activating
+* the `Variable:Created` events create a variables for input mapping
+* the `ServiceTask:Element_Activated` event transitions the service task element to activated
+* the `Job:Created` event creates a job
+* the `JobBatch:Activated` event activates a batch of jobs
+* the `Job:Completed` event completes a job
+* the `Variable:Created` events create variables for output mapping
+* the `Variable:Updated` events update variables for output mapping
+* the `ServiceTask:Element_Completing` event transitions the service task element to completing
+* the `ServiceTask:Element_Completed` event transitions the service task element to completed
+* the `Sequence_Flow_Taken` event marks the sequence flow as taken
+* processing can continue again at `EndEvent:Activate_Element`
+
+Differences to current processing:
+* the event `Job:Created` is produced by the command `ServiceTask:Activate_Element` instead of a `Job:Create` command because it can't be interrupted
+
+Considered alternatives:
+* just like call activity variable propagation we could consider storing job
```

Comment:
Good question. We should describe why. 

Temporary variables in the context of call activity are the variables propagated by the child instance on completion. Output mapping of the call activity element still need to be applied on those before those variables become part of the parent instance's scope. It's an implementation detail for Zeebe to make variable propagation and output mappings work nicely together. We've decided not to introduce TemporaryVariables as records, because it's an implementation detail.

Reply:
@pihme I've added the explanation